system_tests/espresso_e2e_test.go

```diff
@@ -276,6 +275,63 @@ func TestEspressoE2E(t *testing.T) {
 		return balance2.Cmp(transferAmount) >= 0
 	})
 	Require(t, err)
+
+	//  Test that if espresso node is down, the transaction is resubmitted
+	newAccount3 := "User12"
+	l2Info.GenerateAccount(newAccount3)
+	addr3 := l2Info.GetAddress(newAccount3)
+	tx3 := l2Info.PrepareTx("Faucet", newAccount3, 3e7, transferAmount, nil)
+	builder.L1.SendWaitTestTransactions(t, []*types.Transaction{
+		WrapL2ForDelayed(t, tx3, builder.L1Info, "Faucet", 100000),
+	})
```

Comment:
I don't totally understand what's happening here but are we sending this transaction through the L1? In that case is it even going through the sequencer/batcher/espresso? I would have thought L1 transactions get combined with transactions submitted through the L2 _after_ the whole batching process (ie in the derivation pipeline/STF). Entirely possible I'm just misunderstanding.

Bigger concern is that we are sending this transaction in before we shut down Espresso, so might it not already be included before Espresso is shut down? In that case the test wouldn't fail even if we were not resubmitting properly

Reply:
Batcher receives messages from the delayed inbox directly. Good catch with the second point, will update

Reply:
moved it around to shut down the node prior to the delayed inbox transaction. test works fine. resolving this conversation.